lib/thor/shell/basic.rb

```diff
@@ -451,16 +451,22 @@ def ask_simply(statement, color, options)
 
       def ask_filtered(statement, color, options)
         answer_set = options[:limited_to]
+        case_insensitive = options.fetch(:case_insensitive, false)
         correct_answer = nil
         until correct_answer
           answers = answer_set.join(", ")
           answer = ask_simply("#{statement} [#{answers}]", color, options)
-          correct_answer = answer_set.include?(answer) ? answer : nil
+          correct_answer = answer_match(answer_set, answer, case_insensitive)
           say("Your response must be one of: [#{answers}]. Please try again.") unless correct_answer
         end
         correct_answer
       end
 
+      def answer_match(possibilities, answer, case_insensitive)
+        return possibilities.detect{ |possibility| possibility == answer } unless case_insensitive
```

Comment:
can we change this to a `if/else`?

Reply:
You're right; this isn't really a guard clause. I was all about guard clauses when I wrote this.